From 89aa2c3da36d332ffee0198280a72ea0af394b40 Mon Sep 17 00:00:00 2001
From: Rentib <sbitner420@tutanota.com>
Date: Wed, 10 Aug 2022 17:26:38 +0200
Subject: [PATCH] toggleborder

This patch allows you to toggle client border. It works by setting the
border of the client to 0 or to borderpx defined in configuration file.
---
 config.def.h |  1 +
 dwm.c        | 10 ++++++++++
 2 files changed, 11 insertions(+)

diff --git a/config.def.h b/config.def.h
index a2ac963..60b811f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -65,6 +65,7 @@ static Key keys[] = {
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
+  { MODKEY|Mod1Mask|ShiftMask,    XK_b,      toggleborder,   {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
 	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
diff --git a/dwm.c b/dwm.c
index 87d0ada..0e05f30 100644
--- a/dwm.c
+++ b/dwm.c
@@ -211,6 +211,7 @@ static void tag(const Arg *arg);
 static void tagmon(const Arg *arg);
 static void tile(Monitor *m);
 static void togglebar(const Arg *arg);
+static void toggleborder(const Arg *arg);
 static void togglefloating(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
@@ -1705,6 +1706,15 @@ togglebar(const Arg *arg)
 	arrange(selmon);
 }
 
+void
+toggleborder(const Arg *arg)
+{
+  if (selmon && selmon->sel) {
+    selmon->sel->bw = (selmon->sel->bw == borderpx ? 0 : borderpx);
+    arrange(selmon);
+  }
+}
+
 void
 togglefloating(const Arg *arg)
 {
-- 
2.37.1

